### PR TITLE
add to my project my first typescript

### PR DESCRIPTION
**Pull request by Diana064**

### CODE REVIEW

File: src/components/CourseById/Lessons/Lessons.tsx [hunk 1]
```diff
@@ -38,9 +45,10 @@ export const Lessons = ({ course }) => {
         <LessonsListWrapper {...lessonsListWrapperProps} />
       ) : (
         <LessonsList>
-          {lessons.map(lesson => (
-            <LockedLesson key={lesson.id} lesson={lesson} />
-          ))}
+          {lessons &&
+            lessons.map(lesson => (
+              <LockedLesson key={lesson.id} lesson={lesson} />
+            ))}
```
Comment (olesmarola) [inline src/components/CourseById/Lessons/Lessons.tsx:51]:
```suggestion
          {lessons?.map(lesson => (
              <LockedLesson key={lesson.id} lesson={lesson} />
            ))}
```

File: src/components/Courses/Courses.tsx [hunk 2]
```diff
@@ -1,25 +1,39 @@
...
+    courseVideoPreview: string;
+  };
+  previewImageLink: string;
+}
 
 export const Courses = () => {
-  const [courses, setCourses] = useState([]);
-  const [loading, setLoading] = useState(false);
+  const [courses, setCourses] = useState<Course[]>([]);
+  const [loading, setLoading] = useState<boolean>(false);
```
Comment (olesmarola) [inline src/components/Courses/Courses.tsx:31]:
```suggestion
  const [loading, setLoading] = useState(false);
```
unnecessary overtyping

File: src/components/Courses/Courses.tsx [hunk 3]
```diff
@@ -1,25 +1,39 @@
...
+  rating: number;
+  description: string;
+  meta: {
+    skills: string[];
+    courseVideoPreview: string;
+  };
+  previewImageLink: string;
+}
 
 export const Courses = () => {
```
Comment (olesmarola) [inline src/components/Courses/Courses.tsx:29]:
```suggestion
export const Courses: React.FC = () => {
```

File: src/components/Courses/Courses.tsx [hunk 4]
```diff
@@ -1,25 +1,39 @@
...
-  const [courses, setCourses] = useState([]);
-  const [loading, setLoading] = useState(false);
+  const [courses, setCourses] = useState<Course[]>([]);
+  const [loading, setLoading] = useState<boolean>(false);
   const [error, setError] = useState(null);
   const location = useLocation();
   // Variables for pagination
-  const [currentPage, setCurrentPage] = useState(1);
-  const [pageSize] = useState(10);
+  const [currentPage, setCurrentPage] = useState<number>(1);
```
Comment (olesmarola) [inline src/components/Courses/Courses.tsx:35]:
```suggestion
  const [currentPage, setCurrentPage] = useState(1);
```

File: src/components/Courses/Courses.tsx [hunk 5]
```diff
@@ -1,25 +1,39 @@
...
+  const [pageSize] = useState<number>(10);
```
Comment (olesmarola) [inline src/components/Courses/Courses.tsx:36]:
```suggestion
  const [pageSize] = useState(10);
```

File: src/components/HoverVideo/HoverVideo.tsx [hunk 6]
```diff
@@ -1,13 +1,20 @@
...
-  const [hover, setHover] = useState(false);
+interface HoverVideoProps {
+  src: string;
+  img: string;
+  alt: string;
+}
 
-  const handleMouseEnter = () => {
+export const HoverVideo: React.FC<HoverVideoProps> = ({ src, img, alt }) => {
+  const [hover, setHover] = useState<boolean>(false);
```
Comment (olesmarola) [inline src/components/HoverVideo/HoverVideo.tsx:12]:
```suggestion
  const [hover, setHover] = useState(false);
```

File: src/components/Pagination/Pagination.tsx [hunk 7]
```diff
@@ -1,11 +1,23 @@
...
   PrevPage,
   NextPage,
 } from './Pagination.module';
 
-function Pagination({ currentPage, totalPages, setCurrentPage }) {
+type PaginationProps = {
+  currentPage: number;
+  totalPages: number;
+  pageSize: number;
+  setCurrentPage: (currentPage: (page: number) => number) => void;
```
Comment (olesmarola) [inline src/components/Pagination/Pagination.tsx:13]:
```suggestion
  setCurrentPage: Dispatch<number>;
```

File: src/components/Pagination/Pagination.tsx [hunk 8]
```diff
@@ -1,11 +1,23 @@
...
+};
+
+function Pagination({
```
Comment (olesmarola) [inline src/components/Pagination/Pagination.tsx:16]: Why function declaration? You use function expressions for other components.

Comment (olesmarola) [reply src/components/Pagination/Pagination.tsx:16]: And use `React.FC<PaginationProps>`

File: src/components/VideoSpeed/VideoSpeed.tsx [hunk 9]
```diff
@@ -0,0 +1,19 @@
+import { useState, useEffect } from 'react';
```
Comment (olesmarola) [inline src/components/VideoSpeed/VideoSpeed.tsx:1]:
```suggestion
import { useState, useEffect, KeyboardEvent } from 'react';
```

File: src/components/VideoSpeed/VideoSpeed.tsx [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+
+export function usePlaybackRate(initialRate: number): number {
+  const [playbackRate, setPlaybackRate] = useState<number>(initialRate);
+
+  useEffect(() => {
+    function handleKeyDown(event: KeyboardEvent): void {
```
Comment (olesmarola) [inline src/components/VideoSpeed/VideoSpeed.tsx:7]:
```suggestion
    function handleKeyDown(event: KeyboardEvent<Window>): void {
```

File: src/components/hooks/UseProgressTime.tsx [hunk 11]
```diff
@@ -1,21 +1,24 @@
...
+  playedSeconds: number;
+}
 
 export const useProgressTime = () => {
-  const [played, setPlayed] = useLocalStorage('progressTime', {
+  const [played, setPlayed] = useLocalStorage<ProgressTime>('progressTime', {
     playedSeconds: 0,
   });
-  const playerRef = useRef(null);
+  const playerRef = useRef<any>(null);
```
Comment (olesmarola) [inline src/components/hooks/UseProgressTime.tsx:12]: Avoid using `any`

File: src/components/services/api.tsx [hunk 12]
```diff
@@ -4,11 +4,12 @@ axios.defaults.baseURL = 'https://api.wisey.app/api/v1/core/preview-courses';
 const token =
   '?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIzZDJlMzMyMC01Y2I0LTRiMDEtODdiOC00MzVkMzFjMGIzZWUiLCJwbGF0Zm9ybSI6InN1YnNjcmlwdGlvbnMiLCJpYXQiOjE2Nzg4MjYwMDksImV4cCI6MTY3OTcyNjAwOX0.Ij4seua1vaPrlDMkodcNIfGjd4fXbKxHlkxHPj3ts1k';
 
-export const getCourses = async () => {
+export const getCourses = async (): Promise<any[]> => {
```
Comment (olesmarola) [inline src/components/services/api.tsx:7]: Why `any`?

File: src/components/services/api.tsx [hunk 13]
```diff
@@ -4,11 +4,12 @@ axios.defaults.baseURL = 'https://api.wisey.app/api/v1/core/preview-courses';
...
   const response = await axios.get(`${token}`);
   return response.data.courses;
 };
-export const getCoursById = async courseId => {
+
+export const getCoursById = async (courseId: string): Promise<any> => {
```
Comment (olesmarola) [inline src/components/services/api.tsx:12]: Why any?

File: src/pages/CurrentCourse.tsx [hunk 14]
```diff
@@ -1,16 +1,19 @@
...
 import { Loader } from 'components/Loading/Loading';
-export const CurrentCourse = () => {
-  const [course, setCourse] = useState([]);
-  const [error, setError] = useState('');
-  const { courseId } = useParams();
+import { CourseType } from '../types/ComponentsType';
+
+const CurrentCourse = () => {
+  const [course, setCourse] = useState<CourseType | null>(null);
+  const [error, setError] = useState<string>('');
```
Comment (olesmarola) [inline src/pages/CurrentCourse.tsx:12]:
```suggestion
  const [error, setError] = useState('');
```

File: src/pages/CurrentCourse.tsx [hunk 15]
```diff
@@ -1,16 +1,19 @@
...
+  const { courseId } = useParams<{ courseId: string }>();
   const navigate = useNavigate();
   const location = useLocation();
-  const [isLoading, setIsLoading] = useState(false);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
```
Comment (olesmarola) [inline src/pages/CurrentCourse.tsx:16]:
```suggestion
  const [isLoading, setIsLoading] = useState(false);
```

File: src/pages/CurrentCourse.tsx [hunk 16]
```diff
@@ -32,13 +35,13 @@ export const CurrentCourse = () => {
   };
 
   if (!course) {
-    return;
+    return null;
   }
 
   return (
```
Comment (olesmarola) [inline src/pages/CurrentCourse.tsx:41]: Simplify jsx code

File: src/pages/ListCourses.tsx [hunk 17]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import { Courses } from 'components/Courses/Courses';
+
+interface ListCoursesProps {}
```
Comment (olesmarola) [inline src/pages/ListCourses.tsx:4]: remove empty interfaces

**Review by olesmarola — CHANGES_REQUESTED** [review]

**Comment by Diana064** [comment]
change and improve my project thanks

**Pull request merged**